MetaMask/metamask-extension
# Fixing error when user input some letter in token allowance flow

## Explanation

Fixing error when user enter some letter in the input field for token allowance flow. Before this PR, when user enter some letter in the input field, on `develop` MM encountered an error page. With this fix, it is now shows `Error: Enter numbers only` error message and `Next` button is disabled.

## Screenshots/Screencaps

### Before

https://user-images.githubusercontent.com/92527393/211557094-3a2a5974-8fd0-4cc8-bde1-c5010863aaf7.mov

### After

https://user-images.githubusercontent.com/92527393/211557348-f67a4725-8aec-4b95-a283-ad04694095da.mov

## Manual Testing Steps

### Etherscan
1. Unlock MetaMask
2. Go to `Settings -> Experimental`
3. Turn `ON` toggle button for `Improved token allowance experience`
4. Go to https://etherscan.io/token/0x6b175474e89094c44da98b954eedeac495271d0f#writeContract
5. Connect your wallet
6. Under `approve` input any address and a number
7. Click on `write`
8. Enter some letter in the input field 

### Test-Dapp
1. Unlock MetaMask
2. Go to `Settings -> Experimental`
3. Turn `ON` toggle button for `Improved token allowance experience`
4. Open the test dapp https://metamask.github.io/test-dapp/
5. Connect your wallet (click on `Connect` button in `Basic Actions` section)
6. Under the `Send Tokens` section, click on `Create Token`
7. Wait for transaction to be confirmed and click on `Approve Tokens`
or
8. Wait for transaction to be confirmed and click on `Approve Tokens Without Gas`
9. Wait for `Token Allowance` screen is displayed and rendered.
10. Enter some letter in the input field
